dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -12,4 +12,50 @@ def index
   def show
     render json: @section.summarize
   end
+
+  # POST /api/v1/sections
+  # Create a new section
+  def create
+    # Validate login_type
+    # TODO: Push validation into model?
+    login_type = (['none', '', nil].include? params[:login_type]) ? 'email' : params[:login_type]
+    return head :bad_request unless %w(word picture email).include? login_type
+
+    valid_script = params[:script] && valid_script_id?(params[:script][:id])
+    script_to_assign = valid_script && Script.find(params[:script][:id])
+
+    section = Section.create!(
+      {
+        user_id: current_user.id,
+        name: !params[:name].to_s.empty? ? params[:name].to_s : 'New Section',
+        login_type: login_type,
+        grade: Section.valid_grade?(params[:grade].to_s) ? params[:grade].to_s : nil,
+        script_id: script_to_assign ? script_to_assign.id : params[:script_id],
```

Comment:
elsewhere in this method we use `params[:script][:id]` instead of `params[:script_id]`. are we given both?